feature-libs/asm/core/facade/asm-bind-cart.service.ts

```diff
@@ -5,24 +5,45 @@
  */
 
 import { Injectable } from '@angular/core';
-import { AsmBindCartFacade, BindCartParams } from '@spartacus/asm/root';
-import { Command, CommandService } from '@spartacus/core';
-import { Observable } from 'rxjs';
+import { AsmBindCartFacade } from '@spartacus/asm/root';
+import {
+  Command,
+  CommandService,
+  isNotUndefined,
+  UserIdService,
+} from '@spartacus/core';
+import { iif, Observable, of } from 'rxjs';
+import { concatMap } from 'rxjs/operators';
 import { AsmConnector } from '../connectors';
 
 @Injectable()
 export class AsmBindCartService implements AsmBindCartFacade {
   constructor(
     protected commandService: CommandService,
-    protected asmConnector: AsmConnector
+    protected asmConnector: AsmConnector,
+    protected userIdService: UserIdService
   ) {}
 
-  protected bindCartCommand$: Command<BindCartParams> =
-    this.commandService.create((options: BindCartParams) =>
-      this.asmConnector.bindCart(options)
-    );
+  protected bindCartCommand$: Command<
+    { cartId: string; userId?: string },
+    unknown
+  > = this.commandService.create(
+    ({ cartId, userId }: { cartId: string; userId?: string }) =>
```

Comment:
Do we need to accept a userId param?  It seems that what is used by the UI component is the cartId only method signature.  If it is not required, dropping it would simplify the method.